src/ui/bibliography-renderer/index.ts

```diff
@@ -1,39 +1,64 @@
 import joplin from "api";
 import { ContentScriptType } from "api/types";
-import { DataStore } from "../../data/data-store";
-import { fromRefToAPA } from "../../util/apa.util";
-import { REFERENCE_LIST_CONTENT_SCRIPT_ID } from "../../constants";
+import { CSLProcessor } from "../../util/csl-processor";
+import {
+    REFERENCE_LIST_CONTENT_SCRIPT_ID,
+    SETTINGS_CSL_FILE_PATH_ID,
+    MESSAGE_RESTART_APP,
+} from "../../constants";
 
 /**
  * Render the full list of references at the end of the note viewer
  */
-export async function registerBibliographyRenderer (): Promise<void> {
-    
+export async function registerBibliographyRenderer(): Promise<void> {
     /* Register a new content script of type "markdown-it" */
     await joplin.contentScripts.register(
         ContentScriptType.MarkdownItPlugin,
         REFERENCE_LIST_CONTENT_SCRIPT_ID,
-        './ui/bibliography-renderer/render-list-content-script.js'
+        "./ui/bibliography-renderer/render-list-content-script.js"
     );
 
     /**
-     * 1- Lookup reference objects based on their IDs (using DataStore)
-     * 2- Convert reference objects into APA format (in HTML)
-     * If some reference objects are not found, ignore them
-     * No need to html-escape the APA output, the library handles that
+     * Format the references according to the style specified by the user
      */
-    await joplin.contentScripts.onMessage(REFERENCE_LIST_CONTENT_SCRIPT_ID, (IDs: string[]) => {
-        let refs: string[] = [];
-        IDs = [...new Set(IDs)];        // Filter duplicate references
-        IDs.forEach(id => {
-            try {
-                refs.push(`
-                    <li>
-                        ${fromRefToAPA(DataStore.getReferenceById(id))}
-                    </li>
-                `);
-            } catch (e) {}
-        });
-        return refs;
+    const processor = CSLProcessor.getInstance();
+    await joplin.contentScripts.onMessage(
+        REFERENCE_LIST_CONTENT_SCRIPT_ID,
+
+        (IDs: string[]) => {
+            IDs = [...new Set(IDs)]; // Filter duplicate references
+
+            /**
+             * Apply the specified citation style to the references
+             * Note: Does html-encoding by default
+             */
+            return processor.formatRefs(IDs);
+        }
+    );
+    setProcessorStyle(processor);
+
+    /**
+     * Listen to changes applied to the CSL field
+     * On change, set the style of the CSL Processor
+     */
+    joplin.settings.onChange(async (event) => {
+        if (event.keys.includes(SETTINGS_CSL_FILE_PATH_ID)) {
+            await setProcessorStyle(processor);
+            await joplin.views.dialogs.showMessageBox(MESSAGE_RESTART_APP);
```

Comment:
Does that work even from the config screen? Actually, you could simply add a description below the field to say that the app needs to be restarted.